ipaserver/plugins/trustedca.py

```diff
@@ -0,0 +1,199 @@
+from ipalib.parameters import Certificate
+from ipalib.plugable import Registry
+from ipalib import x509
+from ipaserver.plugins.baseldap import (
+    LDAPObject, LDAPSearch, LDAPRetrieve, LDAPUpdate
+)
+from ipaserver.plugins.cert import AbstractCertObject
+from ipapython.dn import DN
+from ipalib import _, ngettext
+
+
+__doc__ = _("""
+Manage trusted Certificate Authorities (ipa-catrust-manage)
+""")
+
+
+register = Registry()
+
+
+@register()
+class trustedca(LDAPObject, AbstractCertObject):
+    """Trusted CA object
+    """
+    container_dn = api.env.container_trustedca
+    object_name = _('Trusted CA certificate')
+    object_name_plural = _('Trusted CAs certificates')
+    object_class = ['ipacertificate']
+    possible_objectclasses = ['pkica', 'ipascopedcertificate']
+    permission_filter_objectclasses = ['ipacertificate']
+    default_attributes = [
+        'cn', 'cacertificate', 'ipakeyextusage', ' ipakeytrust',
+        'ipaCertTrustScope',
+    ]
+    rdn_attribute = 'cn'
+    allow_rename = False
+    label = _('Trusted CA certificates')
+    label_singular = _('Trusted CA certificate')
+
+    takes_params = (
+        Str(
+            'cn',
+            primary_key=True,
+            cli_name='name',
+            label=_('Name'),
+            doc=_('Name for referencing the CA'),
+        ),
+        Str(
+            'ipacertissuerserial',
+            cli_name='issuerserial',
+            label=_('Issuer & Serial'),
+            doc=_('Certificate issuer and serial number'),
+            flags={'no_display', 'no_create', 'no_update', 'no_search'},
+        ),
+        DNParam(
+            'ipacertsubject',
+            cli_name='subject',
+            label=_('Subject DN'),
+            doc=_('Subject Distinguished Name'),
+            flags={'no_display', 'no_create', 'no_update', 'no_search'},
+        ),
+        Bytes(
+            'ipapublickey',
+            label=_("public key"),
+            doc=_("Base-64 encoded public key of the certificate."),
+            flags={'no_display', 'no_create', 'no_update', 'no_search'},
+        ),
+        Certificate(
+            'cacertificate',
+            label=_("CA Certificate"),
+            doc=_("Base-64 encoded certificate."),
+            flags={'no_create', 'no_update', 'no_search'},
+        ),
+    ) + AbstractCertObject.x509v1_params + AbstractCertObject.san_params + (
+        Str(
+            'ipakeyextusage*',
+            cli_name='eku',
+            label=_('Extended key usage'),
+            doc=_('Extended key usage'),
+            flags={'no_create', 'no_update', 'no_search'},
+        ),
+        StrEnum(
+            'ipakeytrust',
+            values=('trusted', 'distrusted'),
+            cli_name='keytrust',
+            label=_('Trust'),
+            doc=_('Trust'),
+            flags={'no_create', 'no_update', 'no_search'},
+        ),
+        StrEnum(
+            'ipacerttrustscope*',
+            values=('http_client_auth', 'pkinit',),
```

Comment:
Does this mean that the CA is not trusted for LDAP (ldaps / STARTTLS), and won't be added to the DS NSSDB?
Or would `tls_client_auth` be a better description?

And should we have another value to indicate that the certificate should be added to the main system trust store?